test/server/config/network/config_test.cc

```diff
@@ -323,11 +319,14 @@ TEST(NetworkFilterConfigTest, BadAccessLogNestedTypes) {
   Json::ObjectSharedPtr json_config = Json::Factory::loadFromString(json_string);
   HttpConnectionManagerFilterConfigFactory factory;
   NiceMock<MockInstance> server;
-  EXPECT_THROW(factory.tryCreateFilterFactory(NetworkFilterType::Read, "http_connection_manager",
-                                              *json_config, server),
+  EXPECT_THROW(factory.createFilterFactory(NetworkFilterType::Read, *json_config, server),
                Json::Exception);
 }
 
+TEST(NetworkFilterConfigTest, DoubleRegistrationTest) {
+  EXPECT_THROW(RegisterNetworkFilterConfigFactory<ClientSslAuthConfigFactory>(), EnvoyException);
```

Comment:
nit: I think EXPECT_THROW_WITH_MESSAGE makes a test slightly better (that macro is from common_test/utility.h)
In Envoy we throw EnvoyExceptions for various reasons it's better to verify that it's what we expect.